### PR TITLE
Fix migration for promulgated url.

### PR DESCRIPTION
**Pull request by fabricematrat**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1800/
Test PASSed.

File: internal/charmstore/migrations.go [hunk 1]
```diff
@@ -398,19 +409,37 @@ func migrateToArchiveDownloadStatsOnly(db StoreDatabase) error {
...
 		}
 	}
 	if err := iter.Err(); err != nil {
 		return errgo.Notef(err, "cannot iterate through entities")
 	}
+
+	for _, promulgated := range  promulgatedBySeriesAndName {
+		legacyDownloadStats, ok := promulgated.ExtraInfo[params.LegacyDownloadStats]
+		if !ok {
+			return errgo.Newf("cannot retrieve the legacy download stats on promulgated: %v", promulgated.URL)
```
Comment (mhilton) [inline internal/charmstore/migrations.go:428]: This should be impossible as we've already got them once. Perhaps add a comment.

**Comment by mhilton** [comment]
:+1:

**Comment by frankban** [comment]
:+1:

**Comment by fabricematrat** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**